core/commands/add.go

```diff
@@ -162,15 +169,15 @@ You can now refer to the added file in a gateway, like so:
 			dserv = dag.NewDAGService(bserv)
 		}
 
-		outChan := make(chan interface{}, 8)
-		res.SetOutput((<-chan interface{})(outChan))
-
-		fileAdder, err := coreunix.NewAdder(req.Context(), n.Pinning, n.Blockstore, dserv)
+		fileAdder, err := coreunix.NewAdder(req.Context(), n.Pinning, n.Blockstore, dserv, cidVer)
 		if err != nil {
 			res.SetError(err, cmds.ErrNormal)
 			return
 		}
 
+		outChan := make(chan interface{}, 8)
```

Comment:
This is not new code I just moved things around a little.  We can do this in a separate p.r.